src/app/static/src/tests/root/actions.test.ts

```diff
@@ -44,5 +44,51 @@ describe("root actions", () => {
 
         actions.validate(mockContext as any);
         expect(mockContext.commit).toHaveBeenCalled();
-    })
+    });
+
+    it("resets state if a state following current step is not valid", () => {
```

Comment:
Took me a minute to understand the name of this test (and the others in the class, some of which I wrote!). Could we just add a comment explaining that "valid" means "either complete or not succeeded by an incomplete step"?

Reply:
It'd be great if you could add that and then merge